### PR TITLE
Hide actual tables from information_schema in Google Sheets

### LINKED ISSUE #2115: Google Sheets connector's information_schema contains wrong tables
It may be specific to gsheets-connector, but `information_schema` schema shows tables from `default` schema as well. It shouldn't.

```
Vikrants-MacBook-Pro:presto-cli-325 vikrant.goel$ ./presto --catalog sheets --schema information_schema
presto:information_schema> show tables;
                     Table
-----------------------------------------------
 applicable_roles
 columns
 enabled_roles
 metadata_table
 number_text
 roles
 schemata
 table_privileges
 table_with_duplicate_and_missing_column_names
 tables
 views
(11 rows)
Query 20191126_194159_00005_m8dx4, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
0:00 [11 rows, 454B] [52 rows/s, 2.12KB/s]

presto:information_schema> select * from number_text;
Query 20191126_194207_00006_m8dx4 failed: line 1:15: Table sheets.information_schema.number_text does not exist
select * from number_text
presto:information_schema>
```

PS. I am using the example google sheet as metadata
https://prestosql.io/docs/current/connector/googlesheets.html
https://docs.google.com/spreadsheets/d/1Es4HhWALUQjoa-bQh4a8B5HROz7dpGMfq_HbfoaW5LM/edit#gid=0

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #2115

### CODE REVIEW

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java [hunk 1]
```diff
@@ -139,6 +140,9 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         if (!schemaName.isPresent()) {
             throw new PrestoException(SHEETS_UNKNOWN_SCHEMA_ERROR, "Schema not present - " + schemaName);
         }
+        if (!listSchemaNames().contains(schemaName.get())) {
```
Comment (findepi) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java]: I think we should explicitly filter out information_schema.

In fact, i think `listTables` should not be called with schemaName=information_schema, but that may require more changes. Currently, each connector needs to take care of this on its own.

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java [hunk 2]
```diff
@@ -139,6 +140,9 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
...
+            return emptyList();
```
Comment (findepi) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java]: ImmutaleList.of

**Comment by kokosing** [comment]
Please add tests that cover #2115.

**Comment by ebyhr** [comment]
Updated. The added test will be failed when we add a new table to inforamtion_schema, though.

File: presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ private static QueryRunner createQueryRunner()
     public void testListTable()
     {
         assertQuery("show tables", "SELECT * FROM (VALUES 'metadata_table', 'number_text', 'table_with_duplicate_and_missing_column_names')");
+        assertQueryReturnsEmptyResult("show tables in gsheets.information_schema like 'number_text'");
```
Comment (findepi) [inline presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java]:
```suggestion
        assertQueryReturnsEmptyResult("SHOW TABLES IN gsheets.information_schema LIKE 'number_text'");
```

**Pull request merged**